### ISSUE TITLE
failures in submodule

### ISSUE THREAD

**Issue by davidmaxwaterman** [tpope/vim-fugitive#769]
set up :

```
[~]$ mkdir tmp
[~]$ cd tmp
[tmp]$ mkdir super
[tmp]$ cd super
[super]$ echo super > super.txt
[super]$ cd ..
[tmp]$ mkdir sub
[tmp]$ cd sub
[sub]$ echo sub > sub.txt
[sub]$ git init
Initialised empty Git repository in /home/davidmaxwaterman/tmp/sub/.git/
[sub]$ git add sub.txt
[sub]$ git commit -a -m 'adding sub.txt'
[master (root-commit) df7c5fa] adding sub.txt
 1 file changed, 1 insertion(+)
 create mode 100644 sub.txt
[sub]$ cd ..
[tmp]$ ls -a
.  ..  sub  super
[tmp]$ cd super
[super]$ ls
super.txt
[super]$ git init
Initialised empty Git repository in /home/davidmaxwaterman/tmp/super/.git/
[super]$ git add super.txt 
[super]$ git commit -a -m 'adding super.txt'
[master (root-commit) b18c1ae] adding super.txt
 1 file changed, 1 insertion(+)
 create mode 100644 super.txt
[super]$ git submodule add ../sub sub
Cloning into 'sub'...
done.
[super]$ cd sub
[sub]$ ls
sub.txt
[sub]$ git log
commit df7c5fa8b892ca8ca495415886df0e84ef254255
Author: Max Waterman <max.waterman@intel.com>
Date:   Mon Mar 21 22:23:17 2016 +0000

    adding sub.txt
[sub]$ vim sub.txt
```

**At this point, alter the text (eg 'This is the submodule.') and do a ':Gdiff'. The resulting diff will be such that the 'old' version is blank when it should contain 'sub'.**

```
[sub]$ git --version
git version 2.7.4
[sub]$ gvim --version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Oct 21 2015 13:03:49)
Included patches: 1-712
```

**Comment by jwhitley** [comment]
@davidmaxwaterman Thanks for the repro steps.  Unfortunately, I'm unable to repro, using vim-fugitive 008b957 or current `master` (e1ae9ef) and:

```
git: 2.7.3
vim: 7.4.1525
 OR
nvim: v0.1.3-331-gd9ecd06
```

For comparison, in `~/tmp/super/sub/.git`, I get the following:
```
> cat .git
gitdir: ../.git/modules/sub
```

What do you have in `.git` from your test case run?

I have *no* idea what would specifically be affected, but I do note that your base vim is pretty old at this point.  Probably also worth checking what vim-fugitive commit you're on.

**Comment by jwhitley** [comment]
(To be clear: @tpope runs the show here.  I'm just a random vim-fugitive user chiming in to help out a bit.)

**Comment by davidmaxwaterman** [comment]
Hi @jwhitley, thanks for your input. You prompted me to check the version of fugutive and it was very old (1.2). I used pathogen method to update it (required me to remove the one I had manually installed in plugin/, in case anyone else stumbles here), and woo hoo, it works.

It's a little odd that the diff is one-on-top-of-the-other, but I guess I can try to fix that.

Thanks again.

/closing.
